qnap/shared/teliod.cgi

```diff
@@ -0,0 +1,126 @@
+    read -n $CONTENT_LENGTH POST_DATA
+fi
+
+QPKG_DIR="/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet"
+APP_TMP_DIR="/tmp/nordsecuritymeshnet"
+APP_CMD="$QPKG_DIR/teliod"
+CONFIG_FILE="$QPKG_DIR/teliod.cfg"
+LOG_FILE="$APP_TMP_DIR/teliod.log"
+PID_FILE="$APP_TMP_DIR/teliod.pid"
+
+is_application_running() {
+    if [ -f ${PID_FILE} ]; then
+        PID=$(cat ${PID_FILE})
+        if [ -d /proc/${PID}/ ]; then
+            echo 1
+        fi
+    fi
+}
+
+send_response() {
+    local code=$1
+    local message=$2
+    echo "{\"code\": $code, \"message\": \"$message\"}"
+    exit $code
+}
+
+start_application() {
+    if [ is_application_running -eq 1 ]; then
+            send_response 400 "Application is already running."
+    fi
+    $APP_CMD daemon $CONFIG_FILE &
+    if [[ $? -eq 0 ]]; then
+        echo $! > ${PID_FILE}
+        send_response 200 "Application started successfully."
+    else
+        echo $! > ${PID_FILE}
+        send_response 500 "Failed to start the application."
+    fi
+}
+
+stop_application() {
+    if [ -f ${TELIOD_PID_FILE} ]; then
+        PID=$(cat ${TELIOD_PID_FILE})
+        kill -9 ${PID} || true
+        rm -f ${TELIOD_PID_FILE}
+        send_response 200 "Application stopped successfully."
+    else
+        send_response 400 "Application is not running."
+    fi
+}
+
+update_config() {
+    NEW_CONFIG=$(echo "$POST_DATA" | jq -r '.config')
+    if [[ -z "$NEW_CONFIG" ]]; then
+        send_response 400 "No configuration provided."
+    fi
+    echo "$NEW_CONFIG" > $CONFIG_FILE
+    if [[ $? -eq 0 ]]; then
+        send_response 200 "Configuration updated successfully."
+    else
+        send_response 500 "Failed to update configuration."
+    fi
+}
+
+get_status() {
+    if [[ -f $PID_FILE ]]; then
+        send_response 200 "Application is running."
+    else
+        send_response 200 "Application is stopped."
+    fi
+}
+
+get_logs() {
+    if [[ -f $LOG_FILE ]]; then
+        LOGS=$(tail -n 100 $LOG_FILE | jq -R -s '.')
+        echo "{\"code\": 200, \"logs\": $LOGS}"
+        exit 0
+    else
+        send_response 404 "Log file not found."
+    fi
+}
+
+case "$REQUEST_METHOD" in
+    POST)
+        if [[ "$QUERY_STRING" == "action=start" ]]; then
+            start_application
+        elif [[ "$QUERY_STRING" == "action=stop" ]]; then
+            stop_application
+        else
+            send_response 400 "Invalid action for POST."
+        fi
+        ;;
+    PATCH)
+        if [[ "$QUERY_STRING" == "action=update-config" ]]; then
+            update_config
+        else
+            send_response 400 "Invalid action for PATCH."
+        fi
+        ;;
+    GET)
+        if [[ "$QUERY_STRING" == "action=get-status" ]]; then
+            get_status
+        elif [[ "$QUERY_STRING" == "action=get-logs" ]]; then
+            get_logs
+        else
+            send_response 400 "Invalid action for GET."
+        fi
+        ;;
+    *)
+       send_response 405 "Method not allowed."
```

Comment:
feels like indentation is off here abit :smile:

Reply:
fixed